src/main/java/com/fauna/client/RequestBuilder.java

```diff
@@ -95,6 +112,35 @@ public HttpRequest buildRequest(Query fql, QueryOptions options, CodecProvider p
         }
     }
 
+    public String buildStreamRequestBody(StreamRequest request) throws IOException {
+        // Use JsonGenerator directly rather than UTF8FaunaGenerator because this is not FQL. For example,
+        // start_ts is a JSON numeric/integer, not a tagged '@long'.
+        ByteArrayOutputStream requestBytes = new ByteArrayOutputStream();
+        JsonGenerator gen = new JsonFactory().createGenerator(requestBytes);
+        gen.writeStartObject();
```

Comment:
Given this all has to be json i'm less confident on the right thing to do as an impl. In your last iteration we def were using the UTF8FaunaGenerator, but in this version it's all JSON, so the StreamRequest class with a basic object mapper (no `@stream` tag it seems?) would be correct and simple.

Reply:
As we discussed, this request is JSON, not FQL, so I'm leaving this as-is for now, and we can improve the docs to make the distinction between the JSON HTTP API and FQL (a subset of JSON).